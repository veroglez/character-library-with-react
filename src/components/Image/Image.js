import React from 'react';
import {
  bool, object, oneOfType, string,
} from 'prop-types';
import assets from '../../assets';
import scss from './Image.module.scss';

const Image = ({ image, icon, ...inherit }) => (
  <div
    className={`${icon ? scss.icon : scss.image}${inherit.styles ? ` ${inherit.styles}` : ''}`}
    style={{ backgroundImage: `url(${image})` }}
  />
);

Image.propTypes = {
  image: oneOfType([object, string]).isRequired,
  icon: bool,
};

Image.defaultProps = {
  icon: undefined,
};

export default Image;
