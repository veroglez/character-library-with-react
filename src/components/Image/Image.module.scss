.image {
  background-size: cover;
  background-position: center;
}

.icon {
  background-size: contain;
  width: 20px;
  height: 20px;
}