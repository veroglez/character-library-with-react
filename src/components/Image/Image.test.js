import React from 'react';
import renderer from 'react-test-renderer';

import Image from './Image';

const IMAGE = 'image.png';
const STYLES = 'styles';

describe('<Image>', () => {
  it('renders when {image}', () => {
    const tree = renderer.create(<Image image={IMAGE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {icon}', () => {
    const tree = renderer.create(<Image icon image={IMAGE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {inherit}', () => {
    const tree = renderer.create(<Image icon image={IMAGE} styles={STYLES} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
