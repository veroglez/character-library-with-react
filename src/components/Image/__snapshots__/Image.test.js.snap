// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image> renders when {image} 1`] = `
<div
  className="undefined"
  style={
    Object {
      "backgroundImage": "url(image.png)",
    }
  }
/>
`;

exports[`<Image> when {icon} 1`] = `
<div
  className="undefined"
  style={
    Object {
      "backgroundImage": "url(image.png)",
    }
  }
/>
`;

exports[`<Image> when {inherit} 1`] = `
<div
  className="undefined styles"
  style={
    Object {
      "backgroundImage": "url(image.png)",
    }
  }
/>
`;
